public/app/features/explore/utils/supplementaryQueries.ts

```diff
@@ -130,7 +130,19 @@ export const getSupplementaryQueryProvider = (
     dsRequest.targets = targets;
 
     if (hasSupplementaryQuerySupport(datasource, type)) {
-      return datasource.getDataProvider(type, dsRequest);
+      if (datasource.getDataProvider) {
+        return datasource.getDataProvider(type, dsRequest);
+      } else if (datasource.getSupplementaryRequest) {
+        const provider = datasource.getSupplementaryRequest(type, dsRequest);
+        if (!provider) {
+          return provider;
+        }
+        return type === SupplementaryQueryType.LogsVolume
+          ? queryLogsVolume(datasource, provider, { targets: dsRequest.targets })
+          : queryLogsSample(datasource, provider);
```

Comment:
```suggestion
        const request = datasource.getSupplementaryRequest(type, dsRequest);
        if (!provider) {
          return undefined;
        }
        return type === SupplementaryQueryType.LogsVolume
          ? queryLogsVolume(datasource, request, { targets: dsRequest.targets })
          : queryLogsSample(datasource, request);
```

Reply:
I think distinguishing between provider and request makes this code more readable.